include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -69,12 +74,66 @@ namespace realization {
                 }
 
+                /**
+                 * Read simulation time from configuration file
+                 */            
+                auto possible_simulation_time = tree.get_child_optional("time");
+
+                if (!possible_simulation_time) {
+                    throw std::runtime_error("ERROR: No simulation time period defined.");
+                }
+
+                geojson::JSONProperty simulation_time_parameters("time", *possible_simulation_time);
+
+                std::vector<std::string> missing_simulation_time_parameters;
+
+                if (!simulation_time_parameters.has_key("start_time")) {
+                    missing_simulation_time_parameters.push_back("start_time");
+                }
+
+                if (!simulation_time_parameters.has_key("end_time")) {
+                    missing_simulation_time_parameters.push_back("end_time");
+                }
+
+                if (!simulation_time_parameters.has_key("output_interval")) {
+                    missing_simulation_time_parameters.push_back("output_interval");
+                }
+
+                if (missing_simulation_time_parameters.size() > 0) {
+                    std::string message = "ERROR: A simulation time parameter cannot be created; the following parameters are missing: ";
+
+                    for (int missing_parameter_index = 0; missing_parameter_index < missing_simulation_time_parameters.size(); missing_parameter_index++) {
+                        message += missing_simulation_time_parameters[missing_parameter_index];
+
+                        if (missing_parameter_index < missing_simulation_time_parameters.size() - 1) {
+                            message += ", ";
+                        }
+                    }
+                    
+                    throw std::runtime_error(message);
+                }
+
+                simulation_time_params simulation_time_config(
+                    simulation_time_parameters.at("start_time").as_string(),
+                    simulation_time_parameters.at("end_time").as_string(),
+                    simulation_time_parameters.at("output_interval").as_natural_number()
+                );
+
+                /**
+                 * Call constructor to construct a Simulation_Time object
+                 */ 
+                this->Simulation_Time_Object = std::make_shared<Simulation_Time>(simulation_time_config);
+
+                /**
+                 * Read catchment configurations from configuration file
+                 */      
                 auto possible_catchment_configs = tree.get_child_optional("catchments");
 
                 if (possible_catchment_configs) {
                     for (std::pair<std::string, boost::property_tree::ptree> catchment_config : *possible_catchment_configs) {
                         this->add_formulation(
                             this->construct_formulation_from_tree(
+                                simulation_time_config,
```

Comment:
The way this exists right now, there is an assumption that output interval is 1-1 with input interval, since it looks like this same config is passed through to the forcing parameters.

Reply:
The simulation_time_config struct is passed to the forcing, but only the start and end times are used. I think that the forcing is actually not currently read? @robertbartel was asking about this yesterday.
https://github.com/jdmattern-noaa/ngen/blob/simulation-time-series/src/NGen.cpp#L229

Reply:
double response = formulation_pair.second->get_response(0, output_time_index, 3600.0, &pdm_et_data);

Reply:
The get_response has calls to get precip from the realization's forcing object, so it is there, just not at the driver level.

Reply:
I think at the very least this needs to be marked with a TODO to separate input_interval from output_interval.

Reply:
I added a TODO there.